forte/data/data_pack.py

```diff
@@ -1597,28 +1619,58 @@ def require_annotations(entry_class=Annotation) -> bool:
             yield from []
             return
 
+        # If range_annotation is specified, we record its begin and
+        # end index
+        range_begin: int
+        range_end: int
+
+        if range_annotation is not None:
+            if isinstance(range_annotation, (Annotation, AudioAnnotation)):
```

Comment:
We can also use `_is_subclass` here to replace `isinstance` checking.

Reply:
For that `range_annotation` should have been a `str`. But we already have `range_annotation` as an object in this case so we can use `isinstance`